pootle/core/views.py

```diff
@@ -49,6 +49,102 @@
 from .utils.json import PootleJSONEncoder
 
 
+class PathRegex(object):
+
+    @property
+    def regex(self):
+        if not (self.project_code or self.language_code):
+            return None
+        regex = r"^/"
+
+        if self.language_code:
+            regex = r"%s%s/" % (regex, self.language_code)
+        else:
+            regex = r"%s[a-zA-Z0-9\-\_]*/" % regex
+
+        if self.project_code:
+            regex = r"%s%s/" % (regex, self.project_code)
+        else:
+            regex = r"%s[a-zA-Z0-9\-\_]*/" % regex
+
+        if self.dir_path:
+            regex = r"%s%s" % (regex, self.dir_path)
+
+        if self.filename:
+            regex = (
+                r"%s%s$"
+                % (regex,
+                   self.filename.replace(".", "\.")))
+        return regex
+
+
+class ViewPathRegex(PathRegex):
+
+    def __init__(self, view):
+        self.view = view
+
+    @property
+    def project_code(self):
+        if self.view.project:
+            return self.view.project.code
+
+    @property
+    def language_code(self):
+        if self.view.language:
+            return self.view.language.code
+
+    @property
+    def filename(self):
+        return self.view.kwargs.get("filename", "")
+
+    @property
+    def dir_path(self):
+        return self.view.kwargs.get("dir_path", "")
+
+
+class RequestPathRegex(PathRegex):
+
+    def __init__(self, path):
+        self.path = path
+
+    @cached_property
+    def kwargs(self):
+        if not self.path:
+            return {}
+        return resolve(self.path).kwargs
+
+    @property
+    def dir_path(self):
+        if not self.path:
+            return ""
+        directory = self.kwargs.get("dir_path", "") or ""
+        if directory and not directory.endswith("/"):
+            directory = "%s/" % directory
+        return directory
+
+    @property
+    def filename(self):
+        return self.kwargs.get("filename", "") or ""
+
+    @property
+    def language_code(self):
+        return self.kwargs.get("language_code", None) or None
+
+    @cached_property
+    def path(self):
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [An attribute defined in pootle.core.views line 108 hides this method](https://www.codacy.com/app/dwayne/pootle/file/1742333681/issues/source?bid=1990595&fileBranchId=3039950#l134)